x-pack/solutions/security/plugins/elastic_assistant/server/routes/knowledge_base/entries/utils.ts

```diff
@@ -24,3 +27,37 @@ export const getKBUserFilter = (user: AuthenticatedUser | null) => {
 
   return `(${globalFilter}${userFilter})`;
 };
+
+export const validateDocumentsModification = async (
+  kbDataClient: AIAssistantKnowledgeBaseDataClient | null,
+  authenticatedUser: AuthenticatedUser | null,
+  documentIds: string[],
+  operation: 'delete' | 'update'
+) => {
+  if (!documentIds.length) {
+    return;
+  }
+  const manageGlobalKnowledgeBaseAIAssistant =
+    kbDataClient?.options?.manageGlobalKnowledgeBaseAIAssistant;
+
+  const userFilter = getKBUserFilter(authenticatedUser);
+  const documentsFilter = documentIds.map((id) => `_id:${id}`).join(' OR ');
+  const entries = await kbDataClient?.findDocuments<EsKnowledgeBaseEntrySchema>({
+    page: 1,
+    perPage: 100,
```

Comment:
Wondering if we should worry about this search at some point. If users will be adding lots of entries etc.